x-pack/plugins/observability_solution/infra/public/containers/metrics_source/metrics_view.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import createContainer from 'constate';
+import { useEffect } from 'react';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import { DEFAULT_METRICS_VIEW_ATTRIBUTES } from '../../../common/constants';
+import { useKibanaContextForPlugin } from '../../hooks/use_kibana';
+import { resolveAdHocDataView } from '../../utils/data_view';
+import { useSourceContext } from './source';
+
+export const useMetricsDataView = () => {
+  const {
+    services: { dataViews },
+  } = useKibanaContextForPlugin();
+
+  const { source } = useSourceContext();
```

Comment:
If we're planning to remove the index pattern from here we could replace this with something that fetches for the index pattern in the metrics data access plugin.  Is that what you're thinking to do in a follow up?

Reply:
exactly, this will be removed once we have a new endpoint to return the index pattern from metrics data access plugin